test/parallel/test-repl.js

```diff
@@ -577,10 +577,13 @@ const errorTests = [
       /^Uncaught Error: Cannot find module 'internal\/repl'/,
       /^Require stack:/,
       /^- <repl>/,
-      /^ {4}at .*/,
-      /^ {4}at .*/,
-      /^ {4}at .*/,
-      /^ {4}at .*/,
+      /^ {4}at .*/, // at Module._resolveFilename
+      /^ {4}at .*/, // at Module._load
+      /^ {4}at .*/, // at node:diagnostics_channel
+      /^ {4}at .*/, // at Channel.runStores
+      /^ {4}at .*/, // at TracingChannel.traceSync
+      /^ {4}at .*/, // at Module.require
+      /^ {4}at .*/, // at require
```

Comment:
I’m glad the function names aren’t in the trace being asserted, but the _number_ of stack frames is, which means that if I add so much as an extra function call to this path within the CommonJS loader then I break this test (just as you did, adding the diagnostics channel). I understand that this test was fragile before, but rather than ignoring the problem (by just increasing the number of frames) it would be better to just solve it, to assert less so that CommonJS loader refactors don’t break this test.

Reply:
Agreed, but I think it's beyond the scope of this PR to define new testing patterns for stack traces. Unless you have a concrete solution to provide I don't think I'll be able to do anything about it here.